mautic/mautic
# Support excel export in reporting, fix double reporting emails [backport]

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ x ]
| New feature/enhancement? (use the a.x branch)      | [ x ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

Reroll of https://github.com/mautic/mautic/pull/10322

I think the best approach is to split this PR up into several as it's partly a bug fix and partly new functionality. For now I'm going to just try and get it working on 4.4 for a client, but we can discuss in the comments how to approach this w.r.t 5.x and the bug fix parts going into 4.4, etc.

Description copied from original PR:

> This pull request adds
>
> * choose format (xlsx,csv) of the report when scheduled  
> * option to not send out blank report  
> * schedule time and timezone of the report to be sent  
>
> In the company there was the request to send out report in XLSX format. So this pull request adds a format dropdown to
> choose format when scheduling report. Also there was a request to not send out report when the report is empty. This was 
> also added as a configuration to the schedule of the report sending out.
>
>While working on that we hit a bug, that sometimes the report was sent out twice a day. The cause was that when scheduling >a report, the report is cheduled first time by web worker in the timezone of the user (in out configuration the Europe/Berlin >UTC+2). When it was sent out by cron, it is then rescheduled in the default timezone of the mautic (in our case UTC). So >example. First time it was scheduled to be sent out at 2021-08-09 00:00:00 CEST (2021-08-08 22:00:00 UTC) and then it was >scheduled to 2021-08-09 00:00:00 UTC, because for the scheduler the next day was not yet reached



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a report
3. Choose schedule and configure new options for the schedule
4. Run a cron to send out a report
